### PR TITLE
Verify kubelet & kube-proxy recovery on Windows nodes

### PR DESCRIPTION
**Pull request by YangLu1031**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:
Add test cases to verify that kubelet & kube-proxy can recover after being killed on windows nodes.
Modify the `test/e2e/apps/daemon_restart.go` to support windows scenario.

**Does this PR introduce a user-facing change?**:

```release-note
Verify kubelet & kube-proxy can recover after being killed on Windows nodes
```

### CODE REVIEW

File: test/e2e/apps/daemon_restart.go [hunk 1]
```diff
@@ -318,4 +325,18 @@ var _ = SIGDescribe("DaemonRestart [Disruptive]", func() {
 			framework.Failf("Net container restart count went from %v -> %v after kubelet restart on nodes %v \n\n %+v", preRestarts, postRestarts, badNodes, tracker)
 		}
 	})
+
+	ginkgo.It("Kube-proxy should recover after being killed accidentally", func() {
```
Comment (pjh) [inline test/e2e/apps/daemon_restart.go]: Did you also run this against a Linux node to make sure it passes?

**Comment by pjh** [comment]
The new code LGTM. I think we'll need to modify the test configs as well to make this test run against Windows in our sig-windows test jobs. I don't remember if that's a k/k change or a test-infra change. Can you look into this?

**Comment by pjh** [comment]
cc @adelina-t @bclau

File: test/e2e/apps/daemon_restart.go (hunk 1 above)
Comment (YangLu1031) [reply test/e2e/apps/daemon_restart.go]: Yeah, just run it, it passed.

**Comment by YangLu1031** [comment]
> The new code LGTM. I think we'll need to modify the test configs as well to make this test run against Windows in our sig-windows test jobs. I don't remember if that's a k/k change or a test-infra change. Can you look into this?

Looks like we already have this test run on sig-windows test jobs: https://testgrid.k8s.io/sig-windows#gce-windows-1.17, "DaemonRestart [Disruptive]"

**Comment by YangLu1031** [comment]
@pjh can you take a look again?

**Comment by pjh** [comment]
We can submit this now and then follow-up with changes to the test configs to get it running continuously for Windows.

/lgtm

@soltysh @kow3ns could you please take a look and approve? This change should be a no-op for existing Linux test runs.

**Comment by YangLu1031** [comment]
ping @soltysh @kow3ns for approval

**Comment by YangLu1031** [comment]
@pjh Changed the 2 tests to `Conformance` test, so that they can show up on `windows-gce-serial` testgrid

**Comment by johnbelamaric** [comment]
Thanks for the PR, but these cannot be conformance tests. These are implementation specifics. There is no reason a given distribution needs to run either of these processes, as long as they can provide the functionality of the cluster without them (kube-proxy in particular is quite replaceable).

**Comment by YangLu1031** [comment]
@pjh changed back to normal test

**Review by janetkuo — APPROVED** [review]

**Pull request merged**